openwisp_users/api/mixins.py

```diff
@@ -206,3 +210,15 @@ class FilterSerializerByOrgOwned(FilterSerializerByOrganization):
     """
 
     _user_attr = 'organizations_owned'
+
+
+class ProtectedAPIMixin(object):
+    """
+    Contains authentication and permission classes for API views
+    """
+
+    authentication_classes = [BearerAuthentication, SessionAuthentication]
+    permission_classes = [
+        IsAuthenticated,
+        DjangoModelPermissions,
+    ]
```

Comment:
let's use tuples instead of lists

Reply:
I think this should be:

```python
permission_classes = (
    IsOrganizationManager,
    DjangoModelPermissions,
)
```

It should be this way because looking at the other modules, they all use this except Network Topology.

Reply:
> I think this should be:
> 
> ```python
> permission_classes = (
>     IsOrganizationManager,
>     DjangoModelPermissions,
> )
> ```
> 
> It should be this way because looking at the other modules, they all use this except Network Topology.
> 
> Once you update this here, please update the other modules to make sure the `permission_classes` property is not redefined.

sure :+1:

Reply:
> ```python
> permission_classes = (
>     IsOrganizationManager,
>     DjangoModelPermissions,
> )
> ```

I have tested `ProtectedAPIMixin` with these permission classes, and they work in every module except `openwisp-users`, See below :

![Screenshot from 2022-12-14 14-12-06](https://user-images.githubusercontent.com/56113566/207548174-b2f1606b-6bcb-43fa-ba46-c23c3c1fc5c2.png)

We need to redefined `permission_classes` property only for `openwisp-users` API views.

Reply:
https://github.com/openwisp/openwisp-users/actions/runs/3692932262/jobs/6252339027#step:11:1

Reply:
Thanks for trying this, I started looking into it and I see that we have a lot of repetition and inconsistencies in the different modules. I think we will have to fix it but I need a bit more time to analyze it and come up with a good solution.